lib/std/crypto/aes.zig

```diff
@@ -9,216 +9,262 @@ const std = @import("../std.zig");
+            var s1 = mem.readIntBig(u32, src[4..8]);
+            var s2 = mem.readIntBig(u32, src[8..12]);
+            var s3 = mem.readIntBig(u32, src[12..16]);
+
+            // First round just XORs input with key.
+            s0 ^= xk[0];
+            s1 ^= xk[1];
+            s2 ^= xk[2];
+            s3 ^= xk[3];
+
+            // Middle rounds shuffle using tables.
+            var k: usize = 4;
+            var t0: u32 = undefined;
+            var t1: u32 = undefined;
+            var t2: u32 = undefined;
+            var t3: u32 = undefined;
+            var r: usize = 0;
+            while (r < numRounds) : (r += 1) {
+                t0 = xk[k + 0] ^ td0[@truncate(u8, s0 >> 24)] ^ td1[@truncate(u8, s3 >> 16)] ^ td2[@truncate(u8, s2 >> 8)] ^ td3[@truncate(u8, s1)];
+                t1 = xk[k + 1] ^ td0[@truncate(u8, s1 >> 24)] ^ td1[@truncate(u8, s0 >> 16)] ^ td2[@truncate(u8, s3 >> 8)] ^ td3[@truncate(u8, s2)];
+                t2 = xk[k + 2] ^ td0[@truncate(u8, s2 >> 24)] ^ td1[@truncate(u8, s1 >> 16)] ^ td2[@truncate(u8, s0 >> 8)] ^ td3[@truncate(u8, s3)];
+                t3 = xk[k + 3] ^ td0[@truncate(u8, s3 >> 24)] ^ td1[@truncate(u8, s2 >> 16)] ^ td2[@truncate(u8, s1 >> 8)] ^ td3[@truncate(u8, s0)];
+                k += 4;
+                s0 = t0;
+                s1 = t1;
+                s2 = t2;
+                s3 = t3;
+            }
+
+            // Last round uses s-box directly and XORs to produce output.
+            s0 = @as(u32, sbox1[t0 >> 24]) << 24 | @as(u32, sbox1[t3 >> 16 & 0xff]) << 16 | @as(u32, sbox1[t2 >> 8 & 0xff]) << 8 | @as(u32, sbox1[t1 & 0xff]);
+            s1 = @as(u32, sbox1[t1 >> 24]) << 24 | @as(u32, sbox1[t0 >> 16 & 0xff]) << 16 | @as(u32, sbox1[t3 >> 8 & 0xff]) << 8 | @as(u32, sbox1[t2 & 0xff]);
+            s2 = @as(u32, sbox1[t2 >> 24]) << 24 | @as(u32, sbox1[t1 >> 16 & 0xff]) << 16 | @as(u32, sbox1[t0 >> 8 & 0xff]) << 8 | @as(u32, sbox1[t3 & 0xff]);
+            s3 = @as(u32, sbox1[t3 >> 24]) << 24 | @as(u32, sbox1[t2 >> 16 & 0xff]) << 16 | @as(u32, sbox1[t1 >> 8 & 0xff]) << 8 | @as(u32, sbox1[t0 & 0xff]);
+
+            s0 ^= xk[k + 0];
+            s1 ^= xk[k + 1];
+            s2 ^= xk[k + 2];
+            s3 ^= xk[k + 3];
+
+            mem.writeIntBig(u32, dst[0..4], s0);
+            mem.writeIntBig(u32, dst[4..8], s1);
+            mem.writeIntBig(u32, dst[8..12], s2);
+            mem.writeIntBig(u32, dst[12..16], s3);
         }
-        pub fn decrypt(ctx: Self, dst: []u8, src: []const u8) void {
-            ctx.dec.decrypt(dst, src);
+
+        // Apply sbox0 to each byte in w.
+        fn subw(w: u32) u32 {
+            return @as(u32, sbox0[w >> 24]) << 24
+                | @as(u32, sbox0[w >> 16 & 0xff]) << 16
+                | @as(u32, sbox0[w >> 8 & 0xff]) << 8
+                | @as(u32, sbox0[w & 0xff]);
         }
-        pub fn ctr(ctx: Self, dst: []u8, src: []const u8, iv: [16]u8) void {
-            ctx.enc.ctr(dst, src, iv);
+
+        fn rotw(w: u32) u32 {
```

Comment:
keep this out of the struct.. or actually; this should be replaced with `std.math.rotl(u32, w, 8)`